full_storage_utilization_test.py

```diff
@@ -0,0 +1,237 @@
+        chunk_size = int(space_needed * 0.1)
+        while current_used < target_used_size and current_usage < target_usage:
+            # Write smaller dataset near the threshold (15% or 30GB of the target)
+            smaller_dataset = (((target_used_size - current_used) < 30) or ((target_usage - current_usage) <= 15))
+            if not smaller_dataset:
+                self.total_large_ks += 1
+            else:
+                self.total_small_ks += 1
+
+            # Use 1GB chunks near threshold, otherwise use 10% of remaining space
+            dataset_size = 1 if smaller_dataset else chunk_size
+            ks_name = "keyspace_small" if smaller_dataset else "keyspace_large"
+            num = self.total_small_ks if smaller_dataset else self.total_large_ks
+            self.log.info(f"Writing chunk of size: {dataset_size} GB")
+            stress_cmd = self.prepare_dataset_layout(dataset_size)
+            stress_queue = self.run_stress_thread(stress_cmd=stress_cmd, keyspace_name=f"{ks_name}{num}", stress_num=1, keyspace_num=num)
+
+            self.verify_stress_thread(cs_thread_pool=stress_queue)
+            self.get_stress_results(queue=stress_queue)
+
+            self.db_cluster.flush_all_nodes()
+            #time.sleep(60) if smaller_dataset else time.sleep(600)
+
+            current_usage, current_used = self.get_max_disk_usage()
+            self.log.info(f"Current max disk usage after writing to keyspace{num}: {current_usage}% ({current_used} GB / {target_used_size} GB)")
+
+    def add_new_node(self):
+        new_nodes = self.db_cluster.add_nodes(count=self.add_node_cnt, enable_auto_bootstrap=True)
+        self.db_cluster.wait_for_init(node_list=new_nodes)
+        self.db_cluster.wait_for_nodes_up_and_normal(nodes=new_nodes)
+        total_nodes_in_cluster = len(self.db_cluster.nodes)
+        self.log.info(f"New node added, total nodes in cluster: {total_nodes_in_cluster}")
+        self.monitors.reconfigure_scylla_monitoring()
+        wait_for_tablets_balanced(self.db_cluster.nodes[0])
+        
+    def remove_node(self):  
+        self.log.info('Removing a second node from the cluster')
+        node_to_remove = self.db_cluster.nodes[1]
+        self.log.info(f"Node to be removed: {node_to_remove.name}")
+        self.db_cluster.decommission(node_to_remove)
+        self.log.info(f"Node {node_to_remove.name} has been removed from the cluster")
+        self.monitors.reconfigure_scylla_monitoring()
+        wait_for_tablets_balanced(self.db_cluster.nodes[0])
+    
+    def get_max_disk_usage(self):
+        max_usage = 0
+        max_used = 0
+        for node in self.db_cluster.nodes:
+            result = node.remoter.run("df -h -BG --output=used,pcent /var/lib/scylla | sed 1d | sed 's/G//' | sed 's/%//'")
+            used, usage = result.stdout.strip().split()
+            max_usage = max(max_usage, int(usage))
+            max_used = max(max_used, int(used))
```

Comment:
```suggestion
            info = self.get_disk_info(node)
            max_usage = max(max_usage, info["used_percent"])
            max_used = max(max_used, info["used"])
```
Since we already have the function that calls `df`

Reply:
updated the code.